### DISCUSSION TITLE
Running out of RAM

### DISCUSSION THREAD

**Discussion by AmbroiseJuston** [idaholab/moose#26746] [Q&A General]
Good afternoon, I am trying to run a 3D mesh (160x40x20 since I had problems with coarser meshes), however my program is almost instantly eating out all my RAM (256Gb available). Does MOOSE have a way to automatically throttle the usage of RAM? I am already throttling back the amount of cores I am using try try and slow things down. I do not have anything to show for it since my computer just dies without much notice.

If it helps i am running on Linux Rocky9.

**Comment by GiudGiud** [comment]
Hello

When does it crash? the mesh is not that big. I think you may be using a non-scalable preconditioniner

Does the live print of the memory consumption tell you which part is consuming so much?

To reduce the memory cost you ll want to use less processes at first.
Using distributed meshes or cheaper preconditioning will let you add more processes

Guillaume

**Reply by AmbroiseJuston** [reply]
The application would usually crash during the adaptation of the mesh. Unfortunately I was not able to find the exact part, simply a correlation of the events.

**Reply by GiudGiud** [reply]
If you turn off adaptivity it goes through?
I think there's an option in the mesh not to repartition. I wonder if the repartitioning is expensive memory wise (of course it helps later on, but there's a solve involved to compute it)

**Reply by AmbroiseJuston** [reply]
Yes removing the adptivity does allow it to run however the 320x80x40 is the largest mesh that can realistically be run. Idealy i was wondering if it was possible to add an option in moose to scale back/slow down as the limits get approached so that instead of crashing the desktop it simply slows itself down. Otherwise i can never create a finer mesh and let it run because it would crash.

**Reply by GiudGiud** [reply]
How many rounds of adaptivity are you running? Maybe you are really running out of memory from adaptivity 

and how many elements are getting refined? Can you limit that?
